src/components/clusters/ClustersTable.tsx

```diff
@@ -64,6 +66,18 @@ const ClustersTable: React.FC<ClustersTableProps> = ({ rows, deleteCluster }) =>
     status: [],
   });
 
+  React.useEffect(() => {
+    const marshalled = window.sessionStorage.getItem(STORAGE_KEY_CLUSTERS_FILTER);
```

Comment:
why not using localStorage? It would allow persisting across tabs. There are no sensitive data stored in the filter so it should be fine, right?

Reply:
The decision was not driven by sensitivity of data but by reseting the state when reentering the application which is expected (I believe). In addition, I do not think we want to persist/share filters among tabs since the user can theoretically use different filters there.

If we want to make the filters persistent between reentering the application, I would leverage the GET URL params for that, so the user can "save" the search. But considering complexity of the recent application, I do not think this would bring significant value to the user.